util.js

```diff
@@ -30,14 +33,24 @@ function isInvalidAsyncPlugin (plugin) {
   return plugin.length !== 2 && plugin.constructor.name === 'AsyncFunction'
 }
 
-function requireServerPluginFromPath (modulePath) {
+async function requireServerPluginFromPath (modulePath) {
   const resolvedModulePath = path.resolve(process.cwd(), modulePath)
 
   if (!fs.existsSync(resolvedModulePath)) {
     throw new Error(`${resolvedModulePath} doesn't exist within ${process.cwd()}`)
   }
 
-  const serverPlugin = require(resolvedModulePath)
+  const modulePattern = /\.mjs$/i
+  let serverPlugin
+  if (modulePattern.test(modulePath)) {
```

Comment:
You also need to load `package.json` and check if `type: 'module'` is there. In that case the `.js` file extension will be used to load modules, and `.cjs` is for esm.

Read more at https://nodejs.org/api/packages.html#packages_package_json_and_file_extensions

Reply:
I added the use case as requested including a test case.

One more question though:

I think its quite inconvenient that each and every project that uses plugins would need to added this additional complexity. 
Hence a single module that would abstract this away would be quite convenient imho, e.g. "importAny".
Where `importAny(file)` would conditionally use `import(file)` or `require(file)` depending on the rules depicted in https://nodejs.org/api/packages.html#packages_determining_module_system.

But when I look at "https://nodejs.org/dist/latest-v12.x/docs/api/esm.html#esm_interoperability_with_commonjs" it states:
```quote
An import statement can reference an ES module or a CommonJS module. Other file types such as JSON or native modules are not supported. For those, use module.createRequire().
```
So that would mean that `import()` already does what I would want `importAny` to do.

Although maybe not relevant anymore for fastify-cli, I am still curious to why you don't want to use `import` for CJS even when the docs recommend it for interoperability.

Reply:
Just noticed that I missed the alternative case: package.json says "module" but file says ".cjs"

Reply:
just added the missing use case.

Reply:
It does prove that its a complex thing to do this right.